openshift-kube-apiserver/admission/storage/performantsecuritypolicy/admission.go

```diff
@@ -0,0 +1,176 @@
+	*admission.Handler
+	storagePerformantSecurityPolicyFeatureEnabled bool
+	nsLister                                      corev1listers.NamespaceLister
+}
+
+// SetExternalKubeInformerFactory registers an informer
+func (c *performantSecurityPolicy) SetExternalKubeInformerFactory(kubeInformers informers.SharedInformerFactory) {
+	c.nsLister = kubeInformers.Core().V1().Namespaces().Lister()
+	c.SetReadyFunc(func() bool {
+		return kubeInformers.Core().V1().Namespaces().Informer().HasSynced()
+	})
+}
+
+func (c *performantSecurityPolicy) InspectFeatureGates(featureGates featuregate.FeatureGate) {
+	c.storagePerformantSecurityPolicyFeatureEnabled = featureGates.Enabled(featuregate.Feature(openshiftfeatures.FeatureGateStoragePerformantSecurityPolicy))
+}
+
+func (c *performantSecurityPolicy) ValidateInitialization() error {
+	if c.nsLister == nil {
+		return fmt.Errorf("%s plugin needs a namespace lister", PluginName)
+	}
+	return nil
+}
+
+func (c *performantSecurityPolicy) Admit(ctx context.Context, attributes admission.Attributes, _ admission.ObjectInterfaces) error {
+	if !c.storagePerformantSecurityPolicyFeatureEnabled {
+		return nil
+	}
+
+	if !c.WaitForReady() {
+		return admission.NewForbidden(attributes, fmt.Errorf("not yet ready to handle request"))
+	}
+
+	if attributes.GetResource().GroupResource() != kapi.Resource("pods") ||
+		len(attributes.GetSubresource()) > 0 {
+		return nil
+	}
+
+	pod, ok := attributes.GetObject().(*kapi.Pod)
+	if !ok {
+		return admission.NewForbidden(attributes, fmt.Errorf("unexpected object: %#v", attributes.GetObject()))
+	}
+
+	ns, err := c.nsLister.Get(pod.Namespace)
+	if err != nil {
+		return fmt.Errorf("error listing pod namespace: %v", err)
+	}
+	podNameKey := fmt.Sprintf("%s/%s", attributes.GetName(), attributes.GetNamespace())
+
+	currentFSGroupChangePolicy := extractCurrentFSGroupChangePolicy(pod)
+	if currentFSGroupChangePolicy == nil {
+		currentFSGroupChangePolicy = getDefaultFSGroupChangePolicy(ctx, ns)
+		if currentFSGroupChangePolicy != nil {
+			klog.V(4).Infof("Setting default FSGroupChangePolicy %s for pod %s", *currentFSGroupChangePolicy, podNameKey)
+			audit.AddAuditAnnotations(ctx, "fsGroupChangePolicy", string(*currentFSGroupChangePolicy), "pod", podNameKey)
+			if pod.Spec.SecurityContext != nil {
+				pod.Spec.SecurityContext.FSGroupChangePolicy = currentFSGroupChangePolicy
+			} else {
+				pod.Spec.SecurityContext = &kapi.PodSecurityContext{
+					FSGroupChangePolicy: currentFSGroupChangePolicy,
+				}
+			}
+		}
+	}
+
+	currentSELinuxChangePolicy := extractCurrentSELinuxChangePolicy(pod)
+	if currentSELinuxChangePolicy == nil {
+		currentSELinuxChangePolicy = getDefaultSELinuxChangePolicy(ctx, ns)
+		if currentSELinuxChangePolicy != nil {
+			klog.V(4).Infof("Setting default SELinuxChangePolicy %s for pod %s", *currentSELinuxChangePolicy, podNameKey)
+			audit.AddAuditAnnotations(ctx, "selinuxChangePolicy", string(*currentSELinuxChangePolicy), "pod", podNameKey)
```

Comment:
and here:

`performant-security.openshift.io/selinuxChangePolicy`
`performant-security.openshift.io/pod`

Reply:
same as above.